packages/forklift-console-plugin/src/modules/Providers/views/details/tabs/VirtualMachines/components/VMCellProps.ts

```diff
@@ -1,9 +1,7 @@
 import { ResourceField } from '@kubev2v/common';
 
-import { VmData } from '../ProviderVirtualMachinesRow';
-
-export interface VMCellProps {
-  data: VmData;
+export interface VMCellProps<T> {
+  data: T;
```

Comment:
The generic here is to avoid casts in cell renders. `ProviderVirtualMachine` type is an union type which needs to be limited to the current provider.

Reply:
i.e. 
```
  path: ({ data }) => <TableCell>{(data?.vm as VSphereVM)?.path}</TableCell>,
// vs
path: ({ data }) => <TableCell>{data?.vm?.path}</TableCell>,
```

Reply:
ehh, but you create a lot of not nice code in the cell renderers, so it's moving the dirt from one place to anther ( from the user to the definition )
I prefer keeping the `as ...`  and keeping the cell renderers clean.

we can find a nicer solution for the renderers table that will not require adding unnecessary generics to the renderers definitions, in later time.

Reply:
well, best is to find a solution that will leave the rendering methods clean AND the table nice :-)

but here since cleaning up the `cellRenderers` table is out of this scope, please add the casts, e.g.
`  path: ({ data }) => <TableCell>{(data?.vm as VSphereVM)?.path}</TableCell>,`

and we can find a cleaner way of doing it in a later PR, that will not require adding unnecessary generics to the rendering methods.

Reply:
IMHO generics are a clean solution :)  reverted to casts in the newest force push